### PR TITLE
Spring + Hibernate + JPA + security updates

### PR DESCRIPTION
**Pull request by ericenns**
## Description of changes
Upgraded Spring, Spring Security Oauth2, Hibernate, and Spring Data JPA to their respective latest versions.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/AnalysisOutputFile.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public String getExecutionManagerFileId() {
 	}
 
 	@com.fasterxml.jackson.annotation.JsonIgnore
-	public final ToolExecution getCreatedByTool() {
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/AnalysisOutputFile.java]: Yes all the hibernate getters had to be have final removed as it was causing a hibernate issue.

File: src/main/java/ca/corefacility/bioinformatics/irida/model/export/NcbiBioSampleFiles.java [hunk 2]
```diff
@@ -40,11 +35,15 @@
...
+	@ManyToMany()
+	@JoinTable(joinColumns = @JoinColumn(name = "ncbi_export_biosample_id"))
+	@LazyCollection(LazyCollectionOption.FALSE)
 	private List<SingleEndSequenceFile> files;
 
 	@JsonIgnore
-	@ManyToMany(fetch = FetchType.EAGER)
+	@ManyToMany()
+	@JoinTable(joinColumns = @JoinColumn(name = "ncbi_export_biosample_id"))
+	@LazyCollection(LazyCollectionOption.FALSE)
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/export/NcbiBioSampleFiles.java]: Correct FetchType.EAGER wasn't working for certain relationships so adding @LazyCollection(LazyCollectionOption.FALSE) was the solution. The other option was to change from List to Set type.

Ref: https://stackoverflow.com/questions/25996758/difference-between-lazycollectionlazycollectionoption-false-and-onetomanyfe

File: src/test/resources/ca/corefacility/bioinformatics/irida/ria/web/oauth/RemoteApisIT.xml [hunk 3]
```diff
@@ -22,5 +23,5 @@
 	<client_details_resource_ids
 		client_details_id="1" resource_id="NmlIrida" />
 	<client_details_scope client_details_id="1" scope="read" />
-
+-->
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/test/resources/ca/corefacility/bioinformatics/irida/ria/web/oauth/RemoteApisIT.xml]: Addressed in 230933f

File: src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaWebSecurityConfig.java [hunk 4]
```diff
@@ -197,26 +51,28 @@ public LoginSuccessHandler getLoginSuccessHandler() {
...
+			web.httpFirewall(new DefaultHttpFirewall());
 		}
 
 		@Override
 		protected void configure(HttpSecurity http) throws Exception {
 			authFailureHandler.setDefaultFailureUrl("/login?error=true");
 			// @formatter:off
 			http.requestMatcher(request -> {
-				return !request.getRequestURI().matches("^.*/api.*$");
+				return !request.getRequestURI().matches("^.*/api(?!/oauth)(/.*)?$");
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaWebSecurityConfig.java]: Addressed in 230933f

File: src/main/java/ca/corefacility/bioinformatics/irida/model/export/NcbiBioSampleFiles.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tom114) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/export/NcbiBioSampleFiles.java]: Done in PR #530

**Comment by ericenns** [comment]
There was no other changes you wanted on this right @tom114

**Pull request merged**